mlflow/deployments/utils.py

```diff
@@ -15,3 +19,44 @@ def parse_target_uri(target_uri):
             + "Deployment URIs must be of the form 'target' or 'target:/suffix'"
         )
     return parsed.scheme
+
+
+def _is_valid_uri(uri: str) -> bool:
+    """
+    Evaluates the basic structure of a provided gateway uri to determine if the scheme and
+    netloc are provided
+    """
+    if uri == "databricks":
+        return True
+    try:
+        parsed = urlparse(uri)
+        return parsed.scheme == "databricks" or all([parsed.scheme, parsed.netloc])
+    except ValueError:
+        return False
+
+
+def assemble_uri_path(paths: List[str]) -> str:
+    """
+    Assemble a correct URI path from a list of path parts.
+
+    :param paths: A list of strings representing parts of a URI path.
+    :return: A string representing the complete assembled URI path.
+    """
+    stripped_paths = [path.strip("/").lstrip("/") for path in paths if path]
+    return "/" + posixpath.join(*stripped_paths) if stripped_paths else "/"
+
+
+def resolve_route_url(base_url: str, route: str) -> str:
+    """
+    Performs a validation on whether the returned value is a fully qualified url (as the case
+    with Databricks) or requires the assembly of a fully qualified url by appending the
+    Route return route_url to the base url of the AI Gateway server.
+
+    :param base_url: The base URL. Should include the scheme and domain, e.g.,
+                     ``http://127.0.0.1:6000``.
+    :param route: The route to be appended to the base URL, e.g., ``/api/2.0/gateway/routes/`` or,
```

Comment:
Can we remove gateway from the docstrings?

Reply:
Removed